org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/form/fields/browserfield/BrowserFieldContentHttpResponseInterceptor.java

```diff
@@ -52,4 +59,28 @@ public void intercept(HttpServletRequest req, HttpServletResponse resp) {
       resp.setHeader(HttpServletControl.HTTP_HEADER_CSP, cspToken);
     }
   }
+
+  /**
+   * Returns the "base URI" for the given browser URI. If the argument is {@code null}, {@code null} is returned.
+   * Otherwise, a string is returned that always ends with a {@code "/"} character. Filename, query parameters and
+   * fragment are automatically removed. The result can be used to build absolute URLs pointing to the current
+   * application.
+   * <p>
+   * Examples:
+   * <ul>
+   * <li>http://localhost:8080 → http://localhost:8080/
+   * <li>https://www.example.org/public/my-app/index.html?debug=true → https://www.example.org/public/my-app/
+   * </ul>
+   */
+  @SuppressWarnings("JavadocLinkAsPlainText")
+  protected String toBaseUri(URI browserUri) {
```

Comment:
Maybe move to ClientSession or a helper to make it reusable? Maybe replace last regex with lastIndexOf("/") and substr to make it a little simpler? see OfficeAddinUtil.